source/slang/slang-emit-cpp.cpp

```diff
@@ -394,6 +394,23 @@ SlangResult CPPSourceEmitter::calcTypeName(IRType* type, CodeGenTarget target, S
 {
     switch (type->op)
     {
+        case kIROp_PtrType:
+        {
+            auto ptrType = static_cast<IRPtrType*>(type);
+            SLANG_RETURN_ON_FAIL(calcTypeName(ptrType->getValueType(), target, out));
+            // TODO(JS): It seems although it says it is a pointer, it can actually be output as a reference
+            // not clear where the ptr aspect is there, as in the definition it is just 'out', implying out
+            // is somewhere converted to a ptr?
```

Comment:
At the IR level, there is no semantic difference between pointers and references, and indeed the `RefType` case and any other `IRPtrTypeBase` (there are some other cases that seemingly wouldn't be caught here) should probably all be emitted identically (and that should be as a pointer in the long run).

This is similar to how if you had to translate a function like this:

```c++
void doThings(int& ioVlaue);
```

over to plain C, you'd write it as:

```c
void doThings(int* ioValue);
```

The only reasons to emit things as C++ reference types in our output today are:

1. Because of the current over-simplifications in the emit logic that have been made for HLSL/GLSL which mean we don't properly handle pointer types in the middle of expressions (e.g., we emit `obj.field` instead of `obj->field` or `(*obj).field`).

2. Because we might want to emit C++ code (especially for headers) that follows a more conventional style in terms of making `out` and `inout` parameters use references. (This might matter if there are C++ ABIs where pointers and references aren't actually guaranteed to be binary-compatible in function signatures)